docs/adr/0001_cmsis_component_source_structure_recommendations.md

```diff
@@ -0,0 +1,111 @@
+|               +-- G                   # Source files currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_ARM_MICRO/    # Currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_ARM_STD/      # Currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_GCC_ARM/      # Currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_IAR/          # Currently in rtos/source/TARGET_CORTEX/
+|           +-- mbed_lib.json           # Currently in rtos/source/TARGET_CORTEX/
+|       +-- Core_A/
+|           +-- Include/
+|               +-- H                   # Header files currently in cmsis/TARGET_CORTEX_A/
+|           +-- Source/
+|               +-- I                   # Source files currently in cmsis/TARGET_CORTEX_A/
+|       +-- Core/
+|           +-- Include/
+|               +-- J                   # Header files currently in cmsis/TARGET_CORTEX_M/
+```
+
+However, to cope with Mbed-CLI requirement to find directories to process with the target labels mechanism, some directories will be prefixed with `TARGET_` and capitalized. It will therefore look as follows:
+
+```
++-- rtos/CMSIS_5/
+|   +-- README.md
+|   +-- docs/
+|   +-- CMSIS/
+|       +-- include/
+|               +-- mbed_cmsis_conf.h   # Currently in cmsis/
+|       +-- RTE/
+|           +-- include/
+|               +-- RTE_Components.h    # Currently in cmsis/
+|       +-- RTOS2/
+|           +-- Include/
+|               +-- A                   # Content of rtos/source/TARGET_CORTEX/rtx5/Include/
+|               +-- B                   # Header files currently in rtos/source/TARGET_CORTEX/
+|           +-- RTX/
+|               +-- Config/
+|                   +-- C               # Content of rtos/source/TARGET_CORTEX/rtx5/RTX/Config/
+|               +-- Include/
+|                   +-- D               # Content of rtos/source/TARGET_CORTEX/rtx5/RTX/Include/
+|               +-- Include1/
+|                   +-- cmsis_os.h      # Currently in rtos/source/TARGET_CORTEX/rtx4/
+|               +-- Library/
+|                   +-- cmsis_os1.c     # Currently in rtos/source/TARGET_CORTEX/rtx4/
+|               +-- Source/
+|                   +-- E               # Content of rtos/source/TARGET_CORTEX/rtx5/RTX/Source/
+|           +-- Source/
+|               +-- F                   # Content of rtos/source/TARGET_CORTEX/rtx5/Source
+|               +-- G                   # Source files currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_ARM_MICRO/    # Currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_ARM_STD/      # Currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_GCC_ARM/      # Currently in rtos/source/TARGET_CORTEX/
+|           +-- TOOLCHAIN_IAR/          # Currently in rtos/source/TARGET_CORTEX/
+|           +-- mbed_lib.json           # Currently in rtos/source/TARGET_CORTEX/
+|       +-- TARGET_CORTEX_A/
```

Comment:
`CMSIS_5` repo contains cmsis and rtos2 and other components.

would it make sense to have `CMSIS` here again to have cmsis component?  I can't find better name for the parent repo (currently named CMSIS).

Reply:
The `CMSIS_5` directory tries to follow as closely as possible the directory structure of the upstream repository of the same name. The upstream repository uses the name `CMSIS` and thus this directory uses the same name.